src/Text/Layout/Table/LineStyle.hs

```diff
@@ -100,13 +102,67 @@ asciiVertical NoLine     = ""
+roundedAsciiJoinString :: LineStyle -> LineStyle -> String
+roundedAsciiJoinString h v = roundedAsciiJoinString4 h h v v
+
+-- | Ascii interior joins, allowing the lines to change when passing through the vertex.
+-- Uses pluses for joins. The argument order is west, east, north, then south.
+asciiJoinString4 :: LineStyle -> LineStyle -> LineStyle -> LineStyle -> String
+asciiJoinString4 NoLine NoLine NoLine NoLine          = " "
+asciiJoinString4 NoLine NoLine n      s      | n == s = asciiVertical n
+asciiJoinString4 w      e      NoLine NoLine | w == e = asciiHorizontal w
+asciiJoinString4 w      e      n      s               = aJoins w e n s
+
+-- | Ascii interior joins, allowing the lines to change when passing through the vertex.
+-- Uses rounded joins. The argument order is west, east, north, then south.
+roundedAsciiJoinString4 :: LineStyle -> LineStyle -> LineStyle -> LineStyle -> String
+roundedAsciiJoinString4 NoLine NoLine NoLine NoLine          = " "
+roundedAsciiJoinString4 NoLine NoLine n      s      | n == s = asciiVertical n
+roundedAsciiJoinString4 w      e      NoLine NoLine | w == e = asciiHorizontal w
+roundedAsciiJoinString4 w      e      n      s               = arJoins w e n s
+
+-- | Draw Ascii line joins with pluses. Arguments are in the order west, east,
+-- north, south.
+--
+-- Only 'NoLine', 'SingleLine', and 'DoubleLine' are supported by Ascii. Other
+-- line styles are treated as 'SingleLine'.
+aJoins :: LineStyle -> LineStyle -> LineStyle -> LineStyle -> String
+aJoins _ _ DoubleLine _          = "++"
+aJoins _ _ _          DoubleLine = "++"
+aJoins _ _ _          _          = "+"
+
+-- | Draw Ascii line joins with rounded joins. Arguments are in order west,
+-- east, north, south.
+--
+-- Only 'NoLine', 'SingleLine', and 'DoubleLine' are supported by Ascii. Other
+-- line styles are treated as 'SingleLine'.
+arJoins :: LineStyle -> LineStyle -> LineStyle -> LineStyle -> String
+-- Top joins
+arJoins _          _          NoLine     DoubleLine = ".."
+arJoins _          _          NoLine     _          = "."
+-- Bottom joins
+arJoins _          _          DoubleLine NoLine     = "''"
+arJoins _          _          _          NoLine     = "'"
+-- Left joins
+arJoins NoLine     DoubleLine DoubleLine _          = "::"
+arJoins NoLine     DoubleLine _          _          = ":"
+arJoins NoLine     SingleLine DoubleLine _          = "++"
+arJoins NoLine     SingleLine _          _          = "+"
+-- Right joins
+arJoins DoubleLine NoLine     DoubleLine _          = "::"
+arJoins DoubleLine NoLine     _          _          = ":"
+arJoins SingleLine NoLine     DoubleLine _          = "++"
+arJoins SingleLine NoLine     _          _          = "+"
+-- Interior joins
+arJoins DoubleLine _          DoubleLine _          = "::"
+arJoins DoubleLine _          _          _          = ":"
+arJoins _          _          DoubleLine _          = "++"
+arJoins _          _          _          _          = "+"
```

Comment:
If this can be a separate commit functionally then please make it one.

Reply:
I've separated this code in particular into a separate commit. It's not used at that point, but it doesn't affect compilation or functionality.